src/coreclr/jit/gentree.cpp

```diff
@@ -12495,28 +12495,44 @@ GenTree* Compiler::gtFoldExpr(GenTree* tree)
                 {
                     fgWalkTreePre(&colon_op2, gtClearColonCond);
                 }
-
+#ifdef DEBUG
+                if (verbose)
+                {
+                    printf("\nIdentical GT_COLON trees! ");
+                    gtDispTree(op2);
+                }
+#endif
+                GenTree* op;
                 if (sideEffList == nullptr)
                 {
                     // No side-effects, just return colon_op2
-                    return colon_op2;
+                    JITDUMP("No side effects, bashing to second operand:\n");
+                    op = colon_op2;
                 }
                 else
                 {
 #ifdef DEBUG
                     if (verbose)
                     {
-                        printf("\nIdentical GT_COLON trees with side effects! Extracting side effects...\n");
+                        printf("Extracting side effects...\n");
                         gtDispTree(sideEffList);
-                        printf("\n");
                     }
 #endif
                     // Change the GT_COLON into a GT_COMMA node with the side-effects
                     op2->ChangeOper(GT_COMMA);
                     op2->gtFlags |= (sideEffList->gtFlags & GTF_ALL_EFFECT);
                     op2->AsOp()->gtOp1 = sideEffList;
-                    return op2;
+
+                    JITDUMP("Transformed GT_COLON into GT_COMMA:\n");
+                    op = op2;
                 }
+#ifdef DEBUG
```

Comment:
> One thing I've noticed is that it would potentially be useful to have something like gtDebugDisp* wrappers so that all non-trivial calls wouldn't have to be wrapped in #ifdef DEBUG + if (verbose), which adds quite a bit of noise. Please let me know if that's something desirable (potentially as a future PR) or the existing approach is fine as is.

You can use `DISPTREE(op)` here and `JITDUMP("message")` instead of `ifdef debug \n if (verbose)`

Reply:
> `DISPTREE`

That's very nice, exactly what I was looking for! I will switch to it where applicable in `gentree.cpp`.